backend/src/packages/chaiNNer_standard/image_dimension/utility/get_bounding_box.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import annotations
+
+from typing import Tuple
+
+import numpy as np
+
+from nodes.properties.inputs import ImageInput, SliderInput
+from nodes.properties.outputs import NumberOutput
+from nodes.utils.utils import get_h_w_c
+
+from .. import utility_group
+
+
+@utility_group.register(
+    schema_id="chainner:image:get_bbox",
+    name="Get Bounding Box",
+    description="Gets a bounding box (X, Y, Height, and Width) of the white area of a mask.",
+    icon="BsRulers",
+    inputs=[
+        ImageInput(channels=1),
+        SliderInput(
+            "Threshold",
+            precision=1,
+            minimum=1,
+            controls_step=1,
+            slider_step=1,
+            default=1,
+        ),
+    ],
+    outputs=[
+        NumberOutput("X"),
+        NumberOutput("Y"),
+        NumberOutput("Width"),
+        NumberOutput("Height"),
```

Comment:
I think we should add output types to these. We might not know their exact values, but we know (1) their range and (2) that they will be ints. The type system can then use this information to detect invalid connections between nodes.

About the range, while the range for X and Y is clear (between 0 and image.width-1, and 0 and image.height-1 respectively), I don't know about Width and Height. Can they be 0? Or a more direct question: what happens when no pixels pass the threshold?

```suggestion
        NumberOutput("X", output_type="min(uint, Input0.width - 1)"),
        NumberOutput("Y", output_type="min(uint, Input0.height - 1)"),
        NumberOutput("Width", output_type="min(uint, Input0.width)"),
        NumberOutput("Height", output_type="min(uint, Input0.height)"),
```

Reply:
Ah, I was confused about how the output_types worked. Thanks.
When no pixels pass the threshold, it returns 0 for the x and y and the dimensions of the image. I thought it was better than returning a box with no size, so maybe the output_type should go from 0 to the actual width and height?

Reply:
ah wait no that makes sense I'm dumb

Reply:
> When no pixels pass the threshold, it returns 0 for the x and y and the dimensions of the image

This doesn't really make sense. Conceptually, we should return an empty bounding box. Unfortunately, the empty bounding box doesn't have a unique representation (any BB with width=0 or height=0 is empty), so it's not clear what the values should be.

We could also go the route Crop Content went. It just throws an error if the BB is empty. I think this would make sense here too. Users probably want to use the BB to crop an image, but width=0 and height=0 aren't valid inputs for our Crop nodes.

If we go with throwing an error, then the output type of Width and Height has to be changed to reflect the fact that they won't ever be 0. Use `min(uint, Input0.width) & 1..` for Width and the same with `.height` for Height.